data/api.go

```diff
@@ -0,0 +1,368 @@
+	rtr.HandleFunc("/status", a.GetStatus).Methods("GET")
+	rtr.Handle("/{userID}", varsHandler(a.GetData)).Methods("GET")
+}
+
+func (h varsHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	h(res, req, vars)
+}
+
+func (a *Api) Get501(res http.ResponseWriter, req *http.Request) {
+	res.WriteHeader(501)
+	return
+}
+
+// @Summary Get the api status
+// @Description Get the api status
+// @ID shoreline-user-api-getstatus
+// @Accept  json
+// @Produce  json
+// @Success 200
+// @Failure 500 {string} string "error description"
+// @Router /status [get]
+func (a *Api) GetStatus(res http.ResponseWriter, req *http.Request) {
+	start := time.Now()
+	var s status.ApiStatus
+	if err := a.store.Ping(); err != nil {
+		errorLog := errorStatusCheck.setInternalMessage(err)
+		logError(&errorLog, start)
+		s = status.NewApiStatus(errorLog.Status, err.Error())
+	} else {
+		s = status.NewApiStatus(http.StatusOK, "OK")
+	}
+	if jsonDetails, err := json.Marshal(s); err != nil {
+		jsonError(res, errorLoadingEvents.setInternalMessage(err), start)
+	} else {
+		res.Header().Add("content-type", "application/json")
+		res.WriteHeader(s.Status.Code)
+		res.Write(jsonDetails)
+	}
+	return
+}
+
+// The /data/userId endpoint retrieves device/health data for a user based on a set of parameters
+// userid: the ID of the user you want to retrieve data for
+// uploadId (optional) : Search for Tidepool data by uploadId. Only objects with a uploadId field matching the specified uploadId param will be returned.
+// deviceId (optional) : Search for Tidepool data by deviceId. Only objects with a deviceId field matching the specified uploadId param will be returned.
+// type (optional) : The Tidepool data type to search for. Only objects with a type field matching the specified type param will be returned.
+//					can be /userid?type=smbg or a comma seperated list e.g /userid?type=smgb,cbg . If is a comma seperated
+//					list, then objects matching any of the sub types will be returned
+// subType (optional) : The Tidepool data subtype to search for. Only objects with a subtype field matching the specified subtype param will be returned.
+//					can be /userid?subtype=physicalactivity or a comma seperated list e.g /userid?subtypetype=physicalactivity,steps . If is a comma seperated
+//					list, then objects matching any of the types will be returned
+// startDate (optional) : Only objects with 'time' field equal to or greater than start date will be returned.
+//					Must be in ISO date/time format e.g. 2015-10-10T15:00:00.000Z
+// endDate (optional) : Only objects with 'time' field less than to or equal to start date will be returned.
+//					Must be in ISO date/time format e.g. 2015-10-10T15:00:00.000Z
+// latest (optional) : Returns only the most recent results for each `type` matching the results filtered by the other query parameters
+func (a *Api) GetData(res http.ResponseWriter, req *http.Request, vars map[string]string) {
```

Comment:
Tidepool uses a copy of storage including request Context as a local variable (only for data API route...)
```golang
storageWithCtx := storage.WithContext(req.Context())
// each subsequent calls are not made on storage but on storageWithCtx 
```

Reply:
ok I understand my mistake. I saw that when merging but did not look in the right branch on the TP side (develop Vs. master)

Reply:
I corrected